website/project/model.py

```diff
@@ -291,12 +307,21 @@ def create(cls, auth, **kwargs):
             save=False,
         )
 
+        if (comment.new_mentions):
+            old_mentions = comment.old_mentions
+            validate = lambda m: m not in old_mentions and validate_contributor(m, comment.node.contributors)
+            comment.new_mentions = filter(validate, comment.new_mentions)
+            if len(comment.new_mentions) > 0:
+                project_signals.mention_added.send(comment, auth=auth)
+                old_mentions.extend(comment.new_mentions)
```

Comment:
Sure seems like we should be able to [DRY](https://en.wikipedia.org/wiki/Don%27t_repeat_yourself) this up with the edit case.